Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp

```diff
@@ -423,6 +426,24 @@ namespace AzToolsFramework
             }
         }
 
+        void PrefabIntegrationManager::OnStartPlayInEditorBegin()
+        {
+            // Focus on the root prefab (AZ::EntityId() will default to it)
+            s_prefabFocusPublicInterface->FocusOnOwningPrefab(AZ::EntityId());
+        }
+
+        void PrefabIntegrationManager::OnStopPlayInEditor()
+        {
+            // Refresh all containers when leaving Game Mode to ensure everything is synced.
+            QTimer::singleShot(
+                0,
+                [&]()
+                {
+                    s_containerEntityInterface->RefreshAllContainerEntities(s_editorEntityContextId);
+                }
+            );
```

Comment:
question: Just curious why you can't call `s_containerEntityInterface->RefreshAllContainerEntities(s_editorEntityContextId);` directly? Is it worth stating why we need the `QTimer::singleShot` call?

Reply:
I originally just called it directly, but it would not work. My guess is that this notification is also used to re-enable the Outliner (which is disabled during focus mode), so if it's called before it does not receive the updates in the model.

Reply:
Ah okay, I think a comment to call this out would be valuable otherwise it's just really opaque to the reader

Reply:
Ok, I'll add it in my next PR :)